http/http_service.go

```diff
@@ -1140,3 +1143,44 @@ func (httpSvc *HttpService) healthHandler(c echo.Context) error {
 
 	return c.JSON(http.StatusOK, healthResponse)
 }
+
+func (httpSvc *HttpService) getAutoSwapsHandler(c echo.Context) error {
+	getAutoSwapsResponse, err := httpSvc.api.GetAutoSwaps(c.Request().Context())
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, ErrorResponse{
+			Message: fmt.Sprintf("Failed to get swap settings: %v", err),
+		})
+	}
+
+	return c.JSON(http.StatusOK, getAutoSwapsResponse)
+}
+
+func (httpSvc *HttpService) enableAutoSwapsHandler(c echo.Context) error {
+	var enableAutoSwapsRequest api.EnableAutoSwapsRequest
+	if err := c.Bind(&enableAutoSwapsRequest); err != nil {
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: fmt.Sprintf("Bad request: %s", err.Error()),
+		})
+	}
+
+	err := httpSvc.api.EnableAutoSwaps(c.Request().Context(), &enableAutoSwapsRequest)
```

Comment:
@im-adithya I saw you added a fix https://github.com/getAlby/hub/pull/1270/commits/86e5f47732bf3761165d27867c41784f6b51ff7b but isn't it that we use the HTTP request context rather than the service context if we enable it here?

Reply:
I didn't get you, we can't use the HTTP request context to enable auto-swaps right? Because it has to keep running?

The fix makes ure we pass in a short lived context to make it work, since during the swap we don't have the http context (it is killed as soon as the swaps are enabled) and have to use the context of the swap service.

(I think you got confused because of the same naming, the api.EnableAutoSwaps is different from swap service EnableAutoSwaps method, can you suggest good naming 😬)

Reply:
> The fix makes sure we pass in a short lived context to make it work, since during the swap we don't have the http context (it is killed as soon as the swaps are enabled) and have to use the context of the swap service.

This is the thing - that fix should not be needed if you are using the service context, because it won't be cancelled until the hub is shut down. Can you check you're using the correct context?